src/openforms/forms/json_schema.py

```diff
@@ -179,18 +269,180 @@ def __setitem__(self, key: str, value: JSONValue):
         data = self.data
         key_list = key.split(".")
         for k in key_list[:-1]:
+            assert isinstance(data, dict)
             child = data.get(k, None)
             if child is None:
                 data[k] = {}
             data = data[k]
+        assert isinstance(data, dict)
         data[key_list[-1]] = value
 
-    def __contains__(self, key: str) -> bool:
+    def __contains__(self, key: object) -> bool:
         """Check if a key is present in the internal dictionary. Gets called via
         ``NestedData().get(...)``.
         """
+        assert isinstance(key, str)
         try:
             self[key]
         except KeyError:
             return False
         return True
+
+
+def process_variable_schema_objects_api(
```

Comment:
I would expect this function to live in `openforms.registrations.contrib.objects_api` instead of in the core module, and for it to be looked up/resolved via the registration plugin registry similar to how `plugin.register_submission` is resolved.